capture_service/android_trace_mgr.cc

```diff
@@ -48,6 +50,47 @@ void AndroidTraceManager::TriggerTrace()
     LOGD("Set capture file path as %s", GetTraceFilePath().c_str());
 }
 
+void AndroidTraceManager::TraceByDuration()
+{
+    m_trace_num++;
+    std::string path = kTraceFilePath + "trace";
+    std::string num = std::to_string(m_trace_num);
+    char        full_path[256];
+    sprintf(full_path, "%s-%04u.rd", path.c_str(), m_trace_num);
+    SetCaptureName(path.c_str(), num.c_str());
+    {
+        absl::MutexLock lock(&m_state_lock);
+        m_state = TraceState::Triggered;
+    }
+    SetTraceFilePath(std::string(full_path));
+
+    {
+        absl::MutexLock lock(&m_state_lock);
+        SetCaptureState(1);
```

Comment:
Do we know how safe it is to start and stop the trace while the application is potentially recording or submitting command buffers in another thread?